### PR TITLE
feat(certificates) rework tresor certificate provider

### PR DESCRIPTION
**Pull request by schristoff**
Signed-off-by: Sarah Christoff <sarah.christoff13@gmail.com>

**Description**:
Builds on the work done in https://github.com/openservicemesh/osm/pull/4580 - this removes all of the caching, messaging broker and more to allow Tresor Provider to only focus on certificates.

Please mark with X for applicable areas.

-->
**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| CI System                  | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [X ] |
| Control Plane              | [ ] |
| Demo                       | [ ] |
| Documentation              | [ ] |
| Egress                     | [ ] |
| Ingress                    | [ ] |
| Install                    | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| Performance                | [ ] |
| SMI Policy                 | [ ] |
| Security                   | [ ] |
| Sidecar Injection          | [ ] |
| Tests                      | [ ] |
| Upgrade                    | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?
    no

2. Is this a breaking change?
no

### CODE REVIEW

File: pkg/catalog/helpers_test.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func newFakeMeshCatalogForRoutes(t *testing.T, testParams testParams) *MeshCatal
 
 	stop := make(chan struct{})
 
-	certManager := tresor.NewFakeCertManager(mockConfigurator)
```
Comment (steeling) [inline pkg/catalog/helpers_test.go:56]: so turns out this isn't even used... let's go ahead and remove it from the `MeshCatalog` type altogether, in pkg/catalog/types.go (the certManager field), and all related instances.

File: pkg/certificate/providers/config.go [hunk 2]
```diff
@@ -237,19 +237,21 @@ func (c *Config) getTresorOSMCertificateManager() (certificate.Manager, debugger
...
 	if err != nil {
 		return nil, nil, errors.Errorf("Failed to instantiate Tresor as a Certificate Manager")
 	}
 
-	return certManager, certManager, nil
+	tresorCertManager, err := certificate.NewManager(rootCert, tresorClient, c.cfg.GetServiceCertValidityPeriod(), c.msgBroker)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error instantiating osm certificate.Manager for Vault cert-manager : %w", err)
+	}
+
```
Comment (steeling) [inline pkg/certificate/providers/config.go:253]: remember that you'll need to call:

	`rotor.New(tresorCertManager).Start(checkCertificateExpirationInterval)`

**Review by nojnhuh — APPROVED** [review]

File: pkg/certificate/providers/config.go [hunk 3]
```diff
@@ -237,19 +237,22 @@ func (c *Config) getTresorOSMCertificateManager() (certificate.Manager, debugger
...
-		c.msgBroker,
 	)
 	if err != nil {
 		return nil, nil, errors.Errorf("Failed to instantiate Tresor as a Certificate Manager")
 	}
 
-	return certManager, certManager, nil
+	tresorCertManager, err := certificate.NewManager(rootCert, tresorClient, c.cfg.GetServiceCertValidityPeriod(), c.msgBroker)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error instantiating osm certificate.Manager for Vault cert-manager : %w", err)
```
Comment (jaellio) [inline pkg/certificate/providers/config.go:251]:
```suggestion
		return nil, nil, fmt.Errorf("error instantiating osm certificate.Manager for Tresor cert-manager : %w", err)
```

File: pkg/certificate/providers/tresor/fake.go [hunk 4]
```diff
@@ -5,35 +5,30 @@ import (
...
-}
-
-// NewFakeCertManagerForRotation creates a fake CertManager used for testing certificate rotation
-func NewFakeCertManagerForRotation(cfg configurator.Configurator, msgBroker *messaging.Broker) *CertManager {
-	cm := NewFakeCertManager(cfg)
-	cm.msgBroker = msgBroker
-	return cm
+	tresorCertManager, err := certificate.NewManager(ca, tresorClient, 1*time.Hour, msgBroker)
+	if err != nil {
+		log.Error().Err(err).Msg("error instantiating osm certificate.Manager for Vault cert-manager")
```
Comment (jaellio) [inline pkg/certificate/providers/tresor/fake.go:29]:
```suggestion
		log.Error().Err(err).Msg("error instantiating osm certificate.Manager for Tresor cert-manager")
```

**Review by jaellio — APPROVED** [review]

**Comment by nojnhuh** [comment]
Lint is still failing CI but `make go-lint` is passing for me locally. I'll look into why CI seems broken, but it's unrelated to this PR since `main` is also failing with similar errors.

**Pull request merged**